app/lib/registries/base_registries.rb

```diff
@@ -1,5 +1,5 @@
 module Registries
-  NAMESPACE = 'registries'
+  NAMESPACE = Rails.env.test? ? 'test/registries' : 'registries'
```

Comment:
Maybe instead of changing namespaces, you could use a different cache_store?

so in development/test.rb you set config.cache_store = :mem_cache_store
and production.rb config.cache_store = :file_store (the default)

See: https://guides.rubyonrails.org/caching_with_rails.html section 2.4?